File: backend/app/utils/email.py
"""
Email utility — logs to console in dev (SMTP_USER empty).
In production, configure SMTP_* env vars.
"""
import logging
import smtplib
from email.mime.text import MIMEText

from app.config import settings

logger = logging.getLogger(__name__)


def _send_smtp(to: str, subject: str, body: str) -> None:
    msg = MIMEText(body, "html", "utf-8")
    msg["Subject"] = subject
    msg["From"] = settings.email_from
    msg["To"] = to
    with smtplib.SMTP(settings.smtp_host, settings.smtp_port) as smtp:
        smtp.starttls()
        smtp.login(settings.smtp_user, settings.smtp_pass)
        smtp.sendmail(settings.email_from, [to], msg.as_string())


def send_email(to: str, subject: str, body: str) -> None:
    if not settings.smtp_user:
        logger.info("📧 [DEV EMAIL] to=%s | subject=%s\n%s", to, subject, body)
        return
    try:
        _send_smtp(to, subject, body)
    except Exception as exc:
        logger.error("Failed to send email to %s: %s", to, exc)


def send_verification_email(to: str, token: str) -> None:
    url = f"{settings.frontend_url}/auth/verify-email?token={token}"
    send_email(
        to,
        "【STA 論壇】請驗證您的電子郵件",
        f'<p>請點擊以下連結完成驗證（24 小時內有效）：</p><p><a href="{url}">{url}</a></p>',
    )


def send_password_reset_email(to: str, token: str) -> None:
    url = f"{settings.frontend_url}/auth/reset-password?token={token}"
    send_email(
        to,
        "【STA 論壇】重設您的密碼",
        f'<p>請點擊以下連結重設密碼（1 小時內有效）：</p><p><a href="{url}">{url}</a></p>',
    )

File: backend/src/utils/email.ts
export interface EmailConfig {
  resendApiKey: string
  emailFrom: string
  frontendUrl: string
}

async function sendEmail(to: string, subject: string, html: string, config: EmailConfig): Promise<void> {
  if (!config.resendApiKey) {
    console.log(`[DEV EMAIL] to=${to} | subject=${subject}\n${html}`)
    return
  }
  try {
    const res = await fetch('https://api.resend.com/emails', {
      method: 'POST',
      headers: {
        Authorization: `Bearer ${config.resendApiKey}`,
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        from: config.emailFrom,
        to: [to],
        subject,
        html,
      }),
    })
    if (!res.ok) {
      const err = await res.text()
      console.error(`Failed to send email: ${err}`)
    }
  } catch (e) {
    console.error('Email send error:', e)
  }
}

export async function sendVerificationEmail(to: string, token: string, config: EmailConfig): Promise<void> {
  const url = `${config.frontendUrl}/auth/verify-email?token=${token}`
  await sendEmail(
    to,
    '【STA 論壇】請驗證您的電子郵件',
    `<p>請點擊以下連結完成驗證（24 小時內有效）：</p><p><a href="${url}">${url}</a></p>`,
    config,
  )
}

export async function sendPasswordResetEmail(to: string, token: string, config: EmailConfig): Promise<void> {
  const url = `${config.frontendUrl}/auth/reset-password?token=${token}`
  await sendEmail(
    to,
    '【STA 論壇】重設您的密碼',
    `<p>請點擊以下連結重設密碼（1 小時內有效）：</p><p><a href="${url}">${url}</a></p>`,
    config,
  )
}
